### PR TITLE
CC-1122: add typescript support to dns server, using deno runtime

### PR DESCRIPTION
**Pull request by ryan-gang**

### CODE REVIEW

**Review by rohitpaulk — CHANGES_REQUESTED** [review]
Added notes in the Git PR, dockerfile shouldn't be using app code

**Comment by ryan-gang** [comment]
Seems like Bun is lacking support for UDP sockets.
(https://github.com/oven-sh/bun/issues/1630)

File: compiled_starters/typescript/app/main.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+// You can use print statements as follows for debugging, they'll be visible when running tests.
+console.log("Logs from your program will appear here!");
+
+// Uncomment this block to pass the first stage
+//
+// const udpSocket: dgram.Socket = dgram.createSocket("udp4");
+// udpSocket.bind(2053, "127.0.0.1");
+
+udpSocket.on("message", (data: Buffer, remoteAddr: dgram.RemoteInfo) => {
+    try {
```
Comment (rohitpaulk) [inline compiled_starters/typescript/app/main.ts:12]: @ryan-gang let's make this simpler: 

(a) Do we need a "message" callback? If not, let's remove this
(b) If we need the message callback to keep the server alive, then let's just print out something like "Received data from ${remoteAddr}" here.

File: compiled_starters/typescript/app/main.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        const response = Buffer.from("");
+        udpSocket.send(response, remoteAddr.port, remoteAddr.address);
+    } catch (e) {
+        console.log(`Error receiving data: ${e}`);
+    }
+});
+
+udpSocket.on("error", (err: Error) => {
```
Comment (rohitpaulk) [inline compiled_starters/typescript/app/main.ts:20]: @ryan-gang do we need this? If not let's remove

**Review by rohitpaulk — CHANGES_REQUESTED** [review]
Added notes

File: compiled_starters/typescript/app/main.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+console.log("Logs from your program will appear here!");
+
+// Uncomment this block to pass the first stage
+//
+// const udpSocket: dgram.Socket = dgram.createSocket("udp4");
+// udpSocket.bind(2053, "127.0.0.1");
+//
+// udpSocket.on("message", (data: Buffer, remoteAddr: dgram.RemoteInfo) => {
+//     try {
+//         console.log(`Received data from ${remoteAddr.address}:${remoteAddr.port}`);
```
Comment (rohitpaulk) [inline compiled_starters/typescript/app/main.ts:13]: @ryan-gang can we make this even simpler, maybe just do `console.log(Received...` and not send anything back? That'd help avoid all of this try / catch stuff.

As a side note, the catch block should probably say "error sending data", not "error receiving data".

**Review by rohitpaulk — APPROVED** [review]

File: compiled_starters/typescript/app/main.ts (hunk 3 above)
Comment (ryan-gang) [reply compiled_starters/typescript/app/main.ts:13]: In the first stage the task is to respond back with a UDP packet. So, that part has to stay. 
I have fixed the catch block log.

**Pull request closed without merging**